From 984d213fa8eab8cded29cb3539c79369e71bdf84 Mon Sep 17 00:00:00 2001
From: Sasha Levin <sashal@kernel.org>
Date: Thu, 27 Apr 2023 21:16:45 +0200
Subject: ASoC: amd: yc: Add ASUS M3402RA into DMI table

From: Enrico Belleri <kilgore.trout@idesmi.eu>

[ Upstream commit 110ccfa9dd6ca1272e50fbea614aa31f01d2c17e ]

Fix builtin microphone on ASUS Vivobook S 14 OLED 2022 (M3402RA)

Same issue with this model as apparently with other Rembrandt laptops: https://bugzilla.kernel.org/show_bug.cgi?id=216270

Signed-off-by: Enrico Belleri <kilgore.trout@idesmi.eu>
Link: https://lore.kernel.org/r/20230427191645.24519-1-kilgore.trout@idesmi.eu
Signed-off-by: Mark Brown <broonie@kernel.org>
Stable-dep-of: b24e3590c94a ("ASoC: amd: yc: Fix non-functional mic on ASUS E1504FA")
Signed-off-by: Sasha Levin <sashal@kernel.org>
---
 sound/soc/amd/yc/acp6x-mach.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sound/soc/amd/yc/acp6x-mach.c b/sound/soc/amd/yc/acp6x-mach.c
index 387d228369d43..441b77d00f919 100644
--- a/sound/soc/amd/yc/acp6x-mach.c
+++ b/sound/soc/amd/yc/acp6x-mach.c
@@ -159,6 +159,13 @@ static const struct dmi_system_id yc_acp_quirk_table[] = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "M5402RA"),
 		}
 	},
+	{
+		.driver_data = &acp6x_card,
+		.matches = {
+			DMI_MATCH(DMI_BOARD_VENDOR, "ASUSTeK COMPUTER INC."),
+			DMI_MATCH(DMI_PRODUCT_NAME, "M3402RA"),
+		}
+	},
 	{
 		.driver_data = &acp6x_card,
 		.matches = {
-- 
2.43.0

